rct229/rulesets/ashrae9012019/section19/section19rule1.py

```diff
@@ -0,0 +1,151 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_cooling_type_DX import (
+    is_hvac_sys_cooling_type_dx,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_heating_type_furnace import (
+    is_hvac_sys_heating_type_furnace,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_heating_type_heat_pump import (
+    is_hvac_sys_heating_type_heat_pump,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_one
+from rct229.utils.std_comparisons import std_equal
+
+REQ_HEATING_OVERSIZING_FACTOR = 0.25
+REQ_COOLING_OVERSIZING_FACTOR = 0.15
+
+
+class Section19Rule1(RuleDefinitionListIndexedBase):
+    """Rule 1 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule1, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section19Rule1.HVACRule(),
+            index_rmr="baseline",
+            id="19-1",
+            description="HVAC system coil capacities for the baseline building design shall be oversized by 15% for cooling and 25% for heating.",
+            ruleset_section_title="HVAC - General",
+            standard_section=" Section G3.1.2.2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        hvac_id = find_one(
```

Comment:
can replace this to `find_exactly_one_hvac_system`

Reply:
I leave this part as is since the ```find_exactly_one_hvac_system``` function requires having hvac id.

Reply:
Ok, that is a good point.
I do think the RDS wants us to find out the applicability for each HVAC. `find_one` will just give you the first HVAC system in the `heating_ventilating_air_conditioning_system`.

In that case, I wouldn't put an applicability check here. Instead, I will remove this applicability check.

in this level:
```python
def create_data(self, context, data):
    rmi_b = context.baseline

    hvac_id_to_flags = {
        hvac_id: {
                "is_hvac_sys_heating_type_furnace_flag" : is_hvac_sys_heating_type_furnace(rmi_b, hvac_id),
                "is_hvac_sys_heating_type_heat_pump_flag" : is_hvac_sys_heating_type_heat_pump(rmi_b, hvac_id),
                "is_hvac_sys_cooling_type_dx_flag" : is_hvac_sys_cooling_type_dx(rmi_b, hvac_id),
        }
        for hvac_id in find_all("$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id")
    }
return {
    "hvac_id_to_flags":hvac_id_to_flags
}
```

In the HVAC level,
I will add the `is_applicable` function
```python
def is_applicable(self, context, data=None):
    hvac_b = context.baseline
    hvac_id_b = hvac_b["id"]
    hvac_id_to_flags = data["hvac_id_to_flags"]
    return (
        hvac_id_to_flags[hvac_id_b]["is_hvac_sys_heating_type_furnace_flag"]
        or hvac_id_to_flags[hvac_id_b]["is_hvac_sys_heating_type_heat_pump_flag"]
        or hvac_id_to_flags[hvac_id_b]["is_hvac_sys_cooling_type_dx_flag"]
)
```

Reply:
Thank you so much for the detailed answer Weili. I just addressed this to the code.